### PR TITLE
Support `VARBINARY` query parameter

### PR DESCRIPTION
**Pull request by mdesmet**
## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: tests/integration/test_types_integration.py [hunk 1]
```diff
@@ -121,10 +121,9 @@ def test_char(trino_connection):
 
 def test_varbinary(trino_connection):
     SqlTest(trino_connection) \
-        .add_field(sql="X'65683F'", python='ZWg/') \
-        .add_field(sql="X''", python='') \
-        .add_field(sql="CAST('' AS VARBINARY)", python='') \
-        .add_field(sql="from_utf8(CAST('😂😂😂😂😂😂' AS VARBINARY))", python='😂😂😂😂😂😂') \
```
Comment (hovaesco) [inline tests/integration/test_types_integration.py:127]: Why you removed it?

**Review by hovaesco — APPROVED** [review]
Looks good % one comment

File: tests/integration/test_types_integration.py (hunk 1 above)
Comment (mdesmet) [reply tests/integration/test_types_integration.py:127]: Because `from_utf8` converts into `VARCHAR` so it is not relevant here.

**Review by hovaesco — APPROVED** [review]

**Comment by mdesmet** [comment]
@hashhar: PTAL

File: tests/integration/test_dbapi_integration.py [hunk 2]
```diff
@@ -532,6 +532,15 @@ def test_null_date_with_time_zone(trino_connection):
     assert rows[0][0] is None
 
 
+def test_binary_query_param(trino_connection):
+    cur = trino_connection.cursor(experimental_python_types=True)
+
+    cur.execute("SELECT ?", params=(bytearray("a", "utf-8"),))
```
Comment (hashhar) [inline tests/integration/test_dbapi_integration.py:538]: Also add a case with non-utf-8 bytes (to make sure implementation isn't doing the stupid thing of converting to `str` and then decoding/encoding to byte-array).

**Review by hashhar — APPROVED** [review]
LGTM % additional test

**Review by hashhar — APPROVED** [review]

**Pull request merged**